src/cascade/model/grids.py

```diff
@@ -0,0 +1,174 @@
+"""Support for dismod concepts that are expressed over grids: age-time, smoothings and value priors.
+"""
+from math import isclose, isinf
+
+GRID_SNAP_DISTANCE = 1 / 365
+
+
+class AgeTimeGrid:
+    """Structure for storing age and times over which the model will estimate.
+    Also constructors for making standard age-time grids.
+    """
+
+    @classmethod
+    def uniform(cls, age_start, age_end, age_step, time_start, time_end, time_step):
+        """Construct an age-time grid with uniform spacing on each axis.
+        All values are in units of years.
+
+        Args:
+            age_start: the lowest age in the grid
+            age_end: the highest age in the grid
+            age_step: the spacing of age points
+            time_start: the earliest time in the grid
+            time_end: the latest time in the grid
+            time_step: the spacing of time points
+        """
+        ages = list(range(age_start, age_end, age_step))
+        times = list(range(time_start, time_end, time_step))
+
+        return cls(ages, times)
+
+    def __init__(self, ages, times, snap_distance=GRID_SNAP_DISTANCE):
+        self._ages = tuple(sorted(set(ages)))
+        self._times = tuple(sorted(set(times)))
+        self._snap_distance = snap_distance
+
+    @property
+    def ages(self):
+        return self._ages
+
+    @property
+    def times(self):
+        return self._times
+
+    @property
+    def snap_distance(self):
+        return self._snap_distance
+
+
+def _any_close(value, targets, tolerance):
+    """True if any element of targets is within tolerance of value
+    """
+    for target in targets:
+        if isclose(value, target, abs_tol=tolerance):
+            return True
+    return False
+
+
+def _validate_region(grid, lower_age, upper_age, lower_time, upper_time):
+    """Validates that the described rectangle aligns with the grid.
+    Positive or negative infinite bounds will be treated as stretching to the edge of the grid.
+
+    Raises:
+        ValueError: If the rectangle does not align
+    """
+    if not _any_close(lower_age, grid.ages, grid.snap_distance) and not isinf(lower_age):
+        raise ValueError("Lower age not in underlying grid")
+
+    if not _any_close(upper_age, grid.ages, grid.snap_distance) and not isinf(upper_age):
+        raise ValueError("Upper age not in underlying grid")
+
+    if not _any_close(lower_time, grid.times, grid.snap_distance) and not isinf(lower_time):
+        raise ValueError("Lower time not in underlying grid")
+
+    if not _any_close(upper_time, grid.times, grid.snap_distance) and not isinf(upper_time):
+        raise ValueError("Upper time not in underlying grid")
+
+
+class _RegionView:
```

Comment:
When first reading this name, _RegionView, made me wonder - what is this?
Maybe add a docstring for this class.

Reply:
Sure